Signal/src/contact/OWSContactsManager.m

```diff
@@ -188,14 +200,127 @@ - (void)updateSignalAccounts
         dispatch_async(dispatch_get_main_queue(), ^{
             self.signalAccountMap = [signalAccountMap copy];
             self.signalAccounts = [signalAccounts copy];
+
+            [self updateCachedDisplayNames];
+
             [[NSNotificationCenter defaultCenter]
                 postNotificationName:OWSContactsManagerSignalAccountsDidChangeNotification
                               object:nil];
         });
     });
 }
 
+- (void)updateCachedDisplayNames
+{
+    OWSAssert([NSThread isMainThread]);
+
+    NSMutableDictionary<NSString *, NSString *> *cachedAccountNameMap = [NSMutableDictionary new];
+    NSMutableDictionary<NSString *, NSString *> *cachedFirstNameMap = [NSMutableDictionary new];
+    NSMutableDictionary<NSString *, NSString *> *cachedLastNameMap = [NSMutableDictionary new];
+    for (SignalAccount *signalAccount in self.signalAccounts) {
```

Comment:
If we're backing our displayNameForRecipientId with this cache, and we're only populating the cache with SignalAccount's then it seems like we'd bring back the bug where we don't show the name of ex-signal users who have unregistered.